### PR TITLE
add backstage yaml

### PR DESCRIPTION
**Pull request by moebiusmania**
## Figma reference/Github issue
ArCo meeting follow up
https://www.notion.so/casavo/c57399e600c94a4cba24a424d6ec5b25?pvs=4

## What
adding Backstage catalog config file
https://www.notion.so/casavo/Spotify-Backstage-Usage-89944cf4477d44d6a285d6dbab4635a4?pvs=4

## Why
to enhance the internal software catalog, and to have other projects refer to Habitat as a dependency

### CODE REVIEW

**Comment by moebiusmania** [comment]
@elleuca there are very few info right now since this will be a dependency for other project. Do we want to consider components as APis?

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: habitat
+  description: Casavo Design System
+tags:
+  language: Typescript
```
Comment (elleuca) [inline catalog-info.yaml]:
```suggestion
  - language:typescript
  - npm 
```

this is an array of tags (i.e. strings, with some [specific limits](https://backstage.io/docs/features/software-catalog/descriptor-format#tags-optional))

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: habitat
+  description: Casavo Design System
```
Comment (elleuca) [inline catalog-info.yaml:5]:
```suggestion
metadata:
  name: habitat
  description: Casavo Design System
  annotations:
    github.com/project-slug: casavo/habitat
```

this extra annotation adds some magic linking :)

**Review by elleuca — APPROVED** [review]

**Comment by elleuca** [comment]
@moebiusmania now you can add it though https://backstage.staging.int.casavo.tech/catalog-import ;-)

**Pull request merged**